sha1-file.c

```diff
@@ -1475,7 +1476,7 @@ int oid_object_info_extended(struct repository *r, const struct object_id *oid,
 		}
 
 		/* Check if it is a missing object */
-		if (fetch_if_missing && repository_format_partial_clone &&
+		if (repo.fetch_if_missing && repository_format_partial_clone &&
```

Comment:
This function already accepts a parameter `r` of type `struct repository *`, please use that instead of the non-existing `repo` variable.